api/filter/http/ext_authz.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package envoy.api.v2.filter.http;
+
+import "google/protobuf/duration.proto";
+
+import "validate/validate.proto";
+
+// [#not-implemented-hide:]
+// External Authorization filter calls out to an external service over the
+// gRPC Authorization API defined by api/external_auth.proto.
+// A failed check will cause this filter to return 403 Forbidden.
+message ExtAuthz {
+
+  // The name of the upstream authorization cluster.
+  string cluster_name = 1 [(validate.rules).string.min_bytes = 1];
+
+  // The timeout in milliseconds for the authorization service RPC. If not
+  // set, this defaults to 20ms.
+  google.protobuf.Duration timeout = 2;
```

Comment:
Would it make more sense to have the idea of request timeout expressed at cluster level? This seems like something not specific to the auth filter. @mattklein123

Reply:
@htuch @mattklein123 This timeout is associated with the grpc request. The grpc call could timeout but the grpc server may still be available at the network level.
In this context it's not entirely mandatory and I could just use a default. We can add it once we have more real experience with invoking external calls.
wdyt?

Reply:
When things are coded up, a default timeout will need to be added, so we could hard code something if we want. We can add a custom timeout later. It's probably worth it to add now though. I feel like adding a default timeout on the cluster is something we could do, but I'm not crazy about it honestly since a cluster is orthogonal to a request, and requests to a cluster could have vastly different timeout properties. So I think I lean towards leaving this here, or, adding some kind of common service configuration message that we share amongst all the RPCs like this in Envoy.

Reply:
Duration uses nanosecond as unit, so "timeout in milliseconds" is incorrect.  Just say specifies gRPC timeout would be enough.

Reply:
@mattklein123 sounds good. I will leave it here.